### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by Doraz1**
Just

 created the forwards propagation of a CNN. Spent most of my time reading up on them and I think I get the point. Shouldn't take too long to create the backpropagation and then move on to the evolution engine that takes a generation of nets with fitnesses and creates the next generation out of them.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+#                 new_weights = [w if w <= MAX_WEIGHT else MAX_WEIGHT for w in new_weights]
+#                 new_weights = [w if w >= MIN_WEIGHT else MIN_WEIGHT for w in new_weights]
+#             else:  # give a new random value to weights
+#                 new_weights = np.random.uniform(MIN_WEIGHT, MAX_WEIGHT, len(old_weights))
+#         else:  # keep the old weights
+#             new_weights = old_weights
+#
+#         # TODO : create a net with these new weights and return the net
+#         return new_weights
+
```
Comment (gthecht) [inline main.py]: These can be deleted...

**Comment by gthecht** [comment]
In your commit, just write what is in the commit - what you added and what you removed.
For instance:
Adds EvolutionEngine class that evolves the best networks.

File: main.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    def evolve(self):
+        self.elitism()
+
+        num_survivors = int(SURVIVAL_THRESHOLD * len(self.nets))
+        survivor_list = self.nets[0:num_survivors]
+
+        while len(self.next_gen) < POP_SIZE:
+            index = random.randint(num_survivors)
+            mutated_offspring = self.mutate(survivor_list[index])
```
Comment (gthecht) [inline main.py]: What is self.mutate?

Comment (Doraz1) [reply main.py]: This will take a net that passes the survival threshold, mutate its hyperparameters and add the offspring to the next generation

**Pull request merged**